core/src/mender-artifact.c

```diff
@@ -803,20 +803,21 @@ mender_artifact_read_meta_data(mender_artifact_ctx_t *ctx) {
         return MENDER_FAIL;
     }
 
-    assert(sizeof(size_t) >= sizeof(unsigned long long));
+    assert(sizeof(size_t) >= sizeof(unsigned long));
     const char *start_ptr = ctx->file.name + strlen(prefix);
     char       *end_ptr;
     errno = 0; /* to distinguish between success/failure */
 
-    const size_t index = strtoull(start_ptr, &end_ptr, 10);
+    const size_t index = strtoul(start_ptr, &end_ptr, 10);
     if ((end_ptr == start_ptr)              /* no conversion */
-        || (0 != errno)                     /* out of range (for unsigned long long) */
+        || (0 != errno)                     /* out of range (for unsigned long) */
         || (index >= ctx->payloads.size)) { /* index out of bounds */
         mender_log_error("Invalid artifact format");
         return MENDER_FAIL;
     }
 
-    assert(StringEqual(end_ptr, "/meta-data")); /* just one last sanity check */
+    assert(NULL != end_ptr);
+    assert(0 == strncmp(end_ptr, "/meta-data", 10)); /* just one last sanity check */
```

Comment:
Let's please add `StringEqualN()` and use it here and below. It could be based on a potentially faster `memcmp()` since we apparently cannot rely on `str*` functions stopping on `\0`-bytes anyway...